### ISSUE TITLE
Union of heterogenous tuples type smooshed when unpacked in function call

### ISSUE THREAD

**Issue by layday** [microsoft/pyright#1487]
**Describe the bug**
When a tuple union is unpacked in a function call the first positional parameter inherits the type of all tuple parameters in the union rather than the type of the first parameter in each tuple.

**To Reproduce**
Type check the snippet below.

**Expected behavior**
That no error would be produced.

**Screenshots or Code**
```python
# pyright: strict

from __future__ import annotations

foo = (1, "") if 1 else (None, b"")  # tuple[int, str] | tuple[None, bytes]

def bar(a: int | None, b: str | bytes) -> None:
    ...

# Argument of type "int | str | bytes | None" cannot be assigned to parameter "a" of type "int | None" in function "bar"
#   Type "int | str | bytes | None" cannot be assigned to type "int | None"
#     Type "str" cannot be assigned to type "int | None"
#       "str" is incompatible with "int"
#       Cannot assign to "None"
#     Type "bytes" cannot be assigned to type "int | None"
#       "bytes" is incompatible with "int"
#       Cannot assign to "None"
bar(*foo)

```

**VS Code extension or command-line**
Either, 1.1.110

**Comment by erictraut** [comment]
The code for unpacked arguments is already very complex and involves a bunch of special cases. I added support for unions of same-length tuples, but I don't plan to add additional logic for different-length tuples or unions of tuples and other types.

This will be fixed in the next release.

**Comment by erictraut** [comment]
This is now addressed in Pyright 1.1.111, which I just published. It will also be included in the next release of pylance.
